Vagrantfile

```diff
@@ -21,83 +21,276 @@ Vagrant.configure("2") do |config|
-    set -eux -o pipefail
-    # configuration
-    GO_VERSION="1.13.14"
 
-    # install dnf deps
-    dnf install -y container-selinux gcc git iptables libseccomp-devel lsof make
-
-    # install Go
-    curl -fsSL "https://dl.google.com/go/go${GO_VERSION}.linux-amd64.tar.gz" | tar Cxz /usr/local
-
-    # setup env vars
-    cat >> /etc/environment <<EOF
-PATH=/usr/local/go/bin:$PATH
-GO111MODULE=off
-EOF
-    source /etc/environment
-    cat >> /etc/profile.d/sh.local <<EOF
-GOPATH=\\$HOME/go
-PATH=\\$GOPATH/bin:\\$PATH
-export GOPATH PATH
-EOF
-    source /etc/profile.d/sh.local
-
-    # enter /root/go/src/github.com/containerd/containerd
-    mkdir -p /root/go/src/github.com/containerd
-    ln -s /vagrant /root/go/src/github.com/containerd/containerd
-    cd /root/go/src/github.com/containerd/containerd
+  # Disabled by default. To run:
+  #   vagrant up --provision-with=upgrade-packages
+  # To upgrade only specific packages:
+  #   UPGRADE_PACKAGES=selinux vagrant up --provision-with=upgrade-packages
+  #
+  config.vm.provision "upgrade-packages", type: "shell", run: "never" do |sh|
+    sh.upload_path = "/tmp/vagrant-upgrade-packages"
+    sh.env = {
+        'UPGRADE_PACKAGES': ENV['UPGRADE_PACKAGES'],
+    }
+    sh.inline = <<~SHELL
+        #!/usr/bin/env bash
+        set -eux -o pipefail
+        dnf -y upgrade ${UPGRADE_PACKAGES}
+    SHELL
+  end
 
-    # install runc (or crun) and other components
-    ./script/setup/install-runc
-    ./script/setup/install-cni
-    ./script/setup/install-critools
+  # To re-run, installing CNI from RPM:
+  #   INSTALL_PACKAGES="containernetworking-plugins" vagrant up --provision-with=install-packages
+  #
+  config.vm.provision "install-packages", type: "shell", run: "once" do |sh|
+    sh.upload_path = "/tmp/vagrant-install-packages"
+    sh.env = {
+        'INSTALL_PACKAGES': ENV['INSTALL_PACKAGES'],
+    }
+    sh.inline = <<~SHELL
+        #!/usr/bin/env bash
+        set -eux -o pipefail
+        dnf -y install \
+            container-selinux \
+            curl \
+            gcc \
+            git \
+            iptables \
+            less \
+            libseccomp-devel \
+            libselinux-devel \
+            lsof \
+            make \
+            mercurial \
+            selinux-policy \
+            selinux-policy-devel \
+            selinux-policy-targeted \
+            socat \
+            subversion \
+            vim \
+            wget \
+            ${INSTALL_PACKAGES}
+    SHELL
+  end
 
-    # install containerd
-    make BUILDTAGS="seccomp selinux no_aufs no_btrfs no_devmapper no_zfs" binaries install
+  # To re-run, installing a different version of go:
+  #   GO_VERSION="1.14.6" vagrant up --provision-with=install-golang
+  #
+  config.vm.provision "install-golang", type: "shell", run: "once" do |sh|
+    sh.upload_path = "/tmp/vagrant-install-golang"
+    sh.env = {
+        'GO_VERSION': ENV['GO_VERSION'] || "1.13.14",
+    }
+    sh.inline = <<~SHELL
```

Comment:
Why not just unpack with ` | tar Cxz /usr/local` and set `/etc/environment` as in the current version?

Reply:
So that root can use it without the need to modify the PATH.